lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPass.cpp

```diff
@@ -43,18 +44,29 @@ class StablehloTypeConverter : public TypeConverter {
 
     // TTNN doesn't support either scalars or boolean data. This transformation
     // converts boolean to bfloat16 and scalars to 1-D tensors.
+    // This transformation also convert 64 bit float/integer types to 32 bit
+    // types.
     addConversion([&](RankedTensorType type) -> RankedTensorType {
       bool changed = false;
       Type elementType = type.getElementType();
       llvm::ArrayRef<int64_t> shape = type.getShape();
+      size_t bitWidth = type.getElementTypeBitWidth();
+      MLIRContext *context = elementType.getContext();
       // Convert the element type to bfloat16 if the input is boolean.
-      if (type.getElementTypeBitWidth() == 1) {
-        elementType = BFloat16Type::get(elementType.getContext());
-        changed = true;
-      } else if (type.getElementTypeBitWidth() == 64 &&
-                 isa<IntegerType>(type.getElementType())) {
-        elementType = IntegerType::get(elementType.getContext(), 32);
+      if (bitWidth == 1) {
+        elementType = BFloat16Type::get(context);
         changed = true;
+      } else if (bitWidth == 64) {
+        // Convert 64 bit integer element type to 32 bit integer.
+        if (isa<IntegerType>(type.getElementType())) {
+          elementType = IntegerType::get(context, 32);
+          changed = true;
+        }
+        // Convert 64 bit float element type to 32 bit float.
+        else if (isa<FloatType>(type.getElementType())) {
+          elementType = FloatType::getF32(context);
+          changed = true;
+        }
```

Comment:
This relates to another issue recently filed, I think we want this kind of data format conversion to happen in TTIR.  It's probably sufficient to land this PR as is since there's already a lot of this kind of handling in stablehlo conversion, but at some point we need to make a concerted effort to migrate it to TTIR.

https://github.com/tenstorrent/tt-mlir/issues/1277#issuecomment-2478587644

fyi @mrakitaTT, @AleksKnezevic

Reply:
I agree with you. We need a dedicated TTIR->TTIR pass to handle all the data types issues in one place ultimately. I was planning to work on it but didn't get a chance. We are only missing support for `f64` in stableHLO->TTIR for current tt-torch models. so this is a quick fix till we have a dedicated pass.